41/yilmazhasan/src/db_service.py

```diff
@@ -0,0 +1,264 @@
+from collections import defaultdict
+
+import tweepy
+import os
+
+from flask import Flask
+from flask import render_template
+from flask import request
+from flask import redirect
+from flask_sqlalchemy import SQLAlchemy
+
+from difflib import SequenceMatcher
+
+import pdb
+import models
+import config
+import shared_objects
+
+from contextlib import closing
+import codecs
+import csv
+import ssl
+import sys
+
+import models
+
+import shared_objects
+
+import pdb
+
+try:
+    from urllib.request import urlopen
+except ImportError:
+    from urllib2 import urlopen
+
+
+Tweet = models.Tweet
+TipModel = models.TipModel
+db = shared_objects.db
+
+
+LOCAL_CSV = config.LOCAL_CSV  # for testing
+REMOTE_CSV = config.REMOTE_CSV
+FIELDS = config.FIELDS
+CONTEXT = config.CONTEXT
+TEST = config.TEST
+
+
+class DbService:
+
+    def __init__(self, consumer_key, consumer_secret, access_token, access_token_secret, db):
+
+        self.consumer_key = consumer_key
+        self.consumer_secret = consumer_secret
+        self.access_token = access_token
+        self.access_token_secret = access_token_secret
+
+        self.db = db
+        self.tweets = self.get_tweets()
+        self.tips = self.get_tips()
+
+
+    # ---------- Basic CRUD operations ----------
+
+    def add_tweet(self, tweet):
+        # pdb.set_trace()
+        submittedBefore = self.anySimilarTipSubmittedBefore(tweet.text)
+        if(not submittedBefore) :
+            try:
+                self.db.session.add(tweet)
+                self.db.session.commit()
+                self.tweets.append(tweet)
+                return ""
+            except Exception as exc:
```

Comment:
at least log the exception? now this is no different than not catching the exception